### ISSUE TITLE
Background URLSession with sharedContainerIdentifier

### ISSUE THREAD

**Issue by evgeniyd** [dropbox/SwiftyDropbox#141]
`DropboxTransportClient` creates a background `URLSession`. Without setting up a shared container, requests made with it from Application Extension fail. 

See also [Performing Uploads and Downloads](https://developer.apple.com/library/content/documentation/General/Conceptual/ExtensibilityPG/ExtensionScenarios.html).

> **NOTE**: SwiftyDropbox currently does not support appex.

**Comment by scobbe** [comment]
@evgeniyd: Thanks for reaching out. If we added a constructor flag to `setupWithAppKey` where you could force the SDK to make all requests with a foreground session, would that solve your issue?

**Comment by evgeniyd** [comment]
@scobbe That's definitely would work, as proven by my test where I just hardcoded this behavior. 

But, I was thinking about passing a value to `sharedContainerIdentifier` (which could be `nil` by default) , so the background `URLSession` could be configured with such a `NSURLSessionConfiguration` ([source](https://developer.apple.com/library/content/documentation/General/Conceptual/ExtensibilityPG/ExtensionScenarios.html#//apple_ref/doc/uid/TP40014214-CH21-SW2)). 

I tested it too and it worked! IMO this would be more flexible solution. 

**NOTE:** passing shared contained would presumably also allow to create a shared `Keychain` which is subject of #138
